From 538a266e04633dca6da0104c4639c76bfed7cf94 Mon Sep 17 00:00:00 2001
From: Liu Xiaowen <b37945@freescale.com>
Date: Fri, 22 Nov 2013 16:29:17 +0800
Subject: [PATCH 477/635] ENGR00214328 fix camera taking picture and recording
 can't work on kitkat.

streams data type changed from u8 to i32.

Signed-off-by: Liu Xiaowen <b37945@freescale.com>
---
 mx6/libcamera2/RequestManager.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/mx6/libcamera2/RequestManager.cpp b/mx6/libcamera2/RequestManager.cpp
index 826c0e2..ca5b89f 100755
--- a/mx6/libcamera2/RequestManager.cpp
+++ b/mx6/libcamera2/RequestManager.cpp
@@ -284,7 +284,7 @@ int RequestManager::tryRestartStreams(int requestType)
     bool streamPicture = false;
     bool videoSnapshot = false;
     for (uint32_t i = 0; i < streams.count; i++) {
-        int streamId = streams.data.u8[i];
+        int streamId = streams.data.i32[i];
         if (streamId == STREAM_ID_RECORD) {
             streamRecord = true;
         }
@@ -308,7 +308,7 @@ int RequestManager::tryRestartStreams(int requestType)
     }
 
     for (uint32_t i = 0; i < streams.count; i++) {
-        int streamId = streams.data.u8[i];
+        int streamId = streams.data.i32[i];
         if (!isStreamValid(requestType, streamId, videoSnapshot)) {
             continue;
         }
@@ -336,7 +336,7 @@ int RequestManager::tryRestartStreams(int requestType)
     }
 
     for (uint32_t i = 0; i < streams.count; i++) {
-        int streamId = streams.data.u8[i];
+        int streamId = streams.data.i32[i];
         if (!isStreamValid(requestType, streamId, videoSnapshot)) {
             continue;
         }
-- 
1.8.0

